indexer-app/src/cmr/indexer/services/index_service.clj

```
@@ -697,50 +697,70 @@
+              context index-names (concept-mapping-types concept-type)
+              concept-id revision-id elastic-version elastic-options)
             ;; Index a deleted-granule document when granule is deleted
             (when (= :granule concept-type)
               (let [[tm result] (util/time-execution
-                                 (dg/index-deleted-granule context concept concept-id revision-id elastic-version elastic-options))]
+                                  (dg/index-deleted-granule context concept concept-id revision-id elastic-version elastic-options))]
                 (debug (format "Timed function %s index-deleted-granule took %d ms." (str *ns*) tm))
                 result))
             ;; propagate collection deletion to granules
             (when (= :collection concept-type)
               (let [[tm result] (util/time-execution
-                                 (cascade-collection-delete context concept-mapping-types concept-id revision-id))]
+                                  (cascade-collection-delete context concept-mapping-types concept-id revision-id))]
                 (debug (format "Timed function %s/cascade-collection-delete took %d ms." (str *ns*) tm))
-                result)))))
-      ;; For draft concept, after the index is deleted, remove it from database.
-      (when (cs/is-draft-concept? concept-type)
-        (meta-db/delete-draft context concept)))))
+                result))))))))
+
+(defn- delete-draft-concept
+  [context concept concept-id revision-id options]
+  (when concept
+    (delete-concept-default-helper context concept concept-id revision-id (assoc options :all-revisions-index? true))
+    (delete-concept-default-helper context concept concept-id revision-id (assoc options :all-revisions-index? false))
+    (try
+      (meta-db/delete-draft context concept)
+      (catch Exception e
+        (info (format "Force delete draft ran into exception for concept-id %s due to race condition between indexers
```

why is this not a warning? Does this mean that a deletion is not successful meaning the system is not correct? Would you ever want to filter these out by being in info?

---

This is a message to let us know the race condition occurred with the draft index. It's technically not a warning because we're expecting this race condition and choosing to ignore this error